### PR TITLE
ENH: inject calibration uid

### PR DESCRIPTION
**Pull request by chiahaoliu**
Enhancement on injecting calibration metadata to header. ``calibration_uid`` and ``is_calibration`` filed are now both included in the metadata fields in ``header``. 

Code passes simulation and ``unittest``.

Note: commit history looks long because rebase.

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Does this put the `calibration_uid` into the run_start_header of the calibration itself?

**Comment by chiahaoliu** [comment]
@CJ-Wright 
just push commits, please see the change

File: xpdacq/calib.py [hunk 1]
```diff
@@ -141,12 +144,16 @@ def run_calibration(exposure=60, calibrant_file=None, wavelength=None,
...
     glbl.calib_config_dict = c.ai.getPyFAI()
     Fit2D_dict = c.ai.getFit2D()
     glbl.calib_config_dict.update(Fit2D_dict)
-    glbl.calib_config_dict.update({'file_name': basename})
-    glbl.calib_config_dict.update({'time': timestr})
+    glbl.calib_config_dict.update({'file_name':basename})
+    glbl.calib_config_dict.update({'time':timestr})
+    # FIXME: need a solution for selecting desired calibration image
+    # based on calibration_collection_uid later
+    glbl.calib_config_dict.update({'calibration_collection_uid':
```
Comment (CJ-Wright) [inline xpdacq/calib.py]: Can you also put this into the calibration_dict?

Comment (chiahaoliu) [inline xpdacq/calib.py]: fix in next commit

Comment (CJ-Wright) [inline xpdacq/calib.py]: Awesome, thank you very much for this!

**Comment by CJ-Wright** [comment]
I'm ready for this to be merged.

**Pull request merged**